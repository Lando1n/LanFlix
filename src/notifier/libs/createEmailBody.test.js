const {
  createMovieEmailBody,
  createNewShowEmailBody,
  createRequestEmailBody,
  createShowEmailBody,
} = require("./createEmailBody");

test("Show notification snapshot", () => {
  const name = "Atlanta";
  const body = createShowEmailBody(name);
  expect(body).toMatchSnapshot();
});

test("Movie notification snapshot", () => {
  const name = "Die Hard";
  const body = createMovieEmailBody(name);
  expect(body).toMatchSnapshot();
});

test("New Show notification snapshot", () => {
  const name = "Atlanta";
  const body = createNewShowEmailBody(name);
  expect(body).toMatchSnapshot();
});

test("Request show notification snapshot", () => {
  const name = "Atlanta";
  const mediaType = "show";
  const requester = "Mark";
  const which = "all";
  const body = createRequestEmailBody(name, { mediaType, requester, which });
  expect(body).toMatchSnapshot();
});

test("Request movie notification snapshot", () => {
  const name = "Die Hard";
  const mediaType = "movie";
  const requester = "Mark";
  const body = createRequestEmailBody(name, { mediaType, requester });
  expect(body).toMatchSnapshot();
});
