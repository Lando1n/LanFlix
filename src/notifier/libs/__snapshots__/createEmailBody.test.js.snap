// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie notification snapshot 1`] = `
"Hey there,</br>
<p>
The movie 'Die Hard' has just been added to the Plex Server.</br>
From my understanding, you asked to be notified about this.</br>
<p>
It should be ready any moment to be streamed!
<p>
Don't want this notification? Toggle your subscription on LanFlix."
`;

exports[`New Show notification snapshot 1`] = `
"'Atlanta' appears to be a new show on the server, would you like to subscribe?</br>
<p>
If so, continue to LanFlix (lanflix.firebaseapp.com) and toggle the subscription!"
`;

exports[`Request notification snapshot 1`] = `"Media Type: SHOW</br>SHOW Name: Atlanta</br>Requested by: Mark</br>Which: all"`;

exports[`Show notification snapshot 1`] = `
"Hey there,</br>
<p>
The show 'Atlanta' has just been added to the Plex Server.</br>
From my understanding, you asked to be notified about this.</br>
<p>
It should be ready any moment to be streamed!
<p>
Don't want this notification? Toggle your subscription on LanFlix."
`;
